---
number: 12
date: November 2025
day: 14
---

# Scientific Computing in Rust Monthly #12

Welcome to this month's issue of Scientific Computing in Rust Monthly.

<!-- Editor can write a 2-3 sentence introduction here -->

Scientific Computing in Rust Monthly can be viewed online at [scientificcomputing.rs/monthly](https://scientificcomputing.rs/monthly),
you can sign up to receive an email copy at [mailinglist.scientificcomputing.rs](https://mailinglist.scientificcomputing.rs),
or you can subscribe using the RSS feed at [scientificcomputing.rs/monthly/rss.xml](https://scientificcomputing.rs/monthly/rss.xml).

Items can be added to the next issue of the newsletter by opening a pull request to the
[Scientific Computing in Rust Monthly repository on GitHub](https://github.com/rust-scicomp/scientific-computing-in-rust-monthly).
You can find more information about adding items
[here](https://github.com/rust-scicomp/scientific-computing-in-rust-monthly#contributing-an-item).

## Crate of the month: InterpN
<!--
The crate of the month will be picked by the editor, who will write a brief description of it here.
If you'd like to suggest a crate that we could include here, tell us on the rust-scicomp Zulip chat
(https://rust-scicomp.zulipchat.com/login/).
-->

InterpN provides numerical interpolation for realtime and bare-metal embedded systems. It is available as a Rust crate and as a Python package. InterpN achieves up to 200x speedup when compared to SciPy, a general standard in the scientific community.
More information can be found in [this very recent blog post by James Logan](https://jlogan.dev/blog/2025/11/10/2025-11-10-interpn-fast-interpolation/#).

[crates.io](https://crates.io/crates/interpn) ~ [docs.rs](https://docs.rs/interpn/0.8.1/interpn/) ~ [source](https://github.com/jlogan03/interpn/)

## New releases
<!--
This section can be used to announce new released of libraries. Items should
be sorted in alphabetical order and should use the format:

### [<library name> <release number>](<crates.io link>)
<brief description of the library and its new features in this release>
-->

### [Burn 0.19.1](https://crates.io/crates/burn)

Burn is both a tensor library and a deep learning framework optimized for numerical computing, model inference, and model training. 
The 0.19.0 and 0.19.1 [releases](https://github.com/tracel-ai/burn/releases) bring major improvements to enable efficient distributed training, quantization, and CPU support.

### [interpn v0.8.1](https://crates.io/crates/interpn)

InterpN is a low-latency numerical interpolation library for realtime and bare-metal embedded systems
and scientific computing.

This month's releases significantly improve performance, now surpassing all methods used in Scipy and numpy
including the specialized methods for 1D and 2D across all conditions.

### [ndarray 0.17.1](https://crates.io/crates/ndarray)

`ndarray` provides an n-dimensional container for general elements and numerics in Rust.
This release introduces a new array reference type. 

### [measurements 0.11.1](https://crates.io/crates/measurements)

A lean library to handle physical quantities as typed measurements. Measurements is `[no_std]` compatible and supports 
simple unit conversions and operator overrides to work with measurements directly.
The 0.11.1 release adds support for additional pressure units.

### [Polars 0.52.0](https://crates.io/crates/polars) 

Polars is an analytical query engine written for data frames. Highlights of [this release](https://github.com/pola-rs/polars/releases/tag/rs-0.52.0) 
include the addition of `LazyFrame.{sink,collect}_batches` and deterministic import order for Python Polars package variants.

### [Zarrs 0.22.7](https://crates.io/crates/zarrs)

Zarrs is a Rust library for the Zarr storage format for multidimensional arrays and metadata. 
Mutliple [releases](https://github.com/zarrs/zarrs/releases) add functionality to `Array`, `node`, `Group`, and `DataType`, as well as `hierarchy` module.

## Events
<!--
This section can be used to advertise events. Items should be sorted in date order, with
sooner events first, and should use the format:

### [<event name>, <day-day month year>, <location or "held virutally">, <date>](<website URL>)
<brief description of event>
-->

### [Scientific Computing Track @ Rust NYC UnConf 2025, 5 December 2025, New York City](https://www.meetup.com/rust-nyc/events/311757146/)

Join us to explore the emerging world of scientific computing in Rust. We'll discuss interoperability with other languages, improving accessibility for scientists, best practices, community, and more. We invite you to come and share experiences, challenges, and ideas for shaping this field. Registration, more info, and a complete list of tracks are available at https://www.meetup.com/rust-nyc/events/311757146/

## Publications
<!--
This section can be used to publicise papers, articles and blog posts published about scientific computing in Rust.
Items should be sorted in the alphabetical order of the surname of the first listed author.
Papers posted should use the following format:

### [<authors>, <title>, <journal> <volume>](<DOI>)
<brief description of paper>

Other publications should use the following format:

### [<author>, <title>](<link>)
<brief description>
-->

### [Heim G. M., GlobalSearch-rs: A multistart framework for global optimization written in Rust](https://joss.theoj.org/papers/10.21105/joss.09234)

This paper presents `GlobalSearch-rs`, an open-source library written in Rust that implements 
a multistart framework for global optimization of NLP problems.

### [Hovey C. B. and Buche M. R., automesh: Automatic mesh generation in Rust](https://joss.theoj.org/papers/10.21105/joss.08768)

This paper presents `automesh`, an open-source Rust software program that uses a segmentation that is typically generated
from a 3D image stack to create a finite element mesh composed either of hexahedral
(volumetric) or triangular (isosurface) elements.

### [Logan J., InterpN: Fast Interpolation](https://jlogan.dev/blog/2025/11/10/2025-11-10-interpn-fast-interpolation/)

A blog post describing the algorithms and performance engineering behind `interpn`, with
reusable patterns applicable to almost any high-performance Rust or Rust-Python scientific computing library.

Come for the compile-time loop unrolling, stay for the profile-guided optimization!


### [Scheidwasser N., et al., phylo2vec: a library for vector-based phylogenetic tree manipulation](https://joss.theoj.org/papers/10.21105/joss.09040)

This paper presents the latest release of `phylo2vec`, a high-performance software
package for encoding, manipulating, and analyzing binary phylogenetic trees. 

### [St√∂ckli L. L., et al., THz Image Explorer - An Interactive Cross-Platform

Open-Source THz Image Analysis Tool](https://joss.theoj.org/papers/10.21105/joss.08754)
This paper presents a ThZ time-domain spectroscopy analysis tool for interactive data analysis written in Rust.

## Help wanted
<!--
If you're looking for people to help develop or maintain a project, you can add it here. Items here
should be sorted in alphabetical order by project name and use the format:

### [<library name>](<link to source code repository>)
<brief description of the library and what kind of help you're interested in>
-->

### [Seq.rs](https://github.com/dawnandrew100/seq.rs)

Seq.rs is a website that is a showcase of the work being done in bioinformatics using Rust. Each month, there is a theme with related crates, a research paper summary, and a monthly coding challenge.

I specifically need help with coming up with coding challenges, which will then be posted to the [companion GitHub](https://github.com/dawnandrew100/seqrs-challenges), but help with any part of the project would be greatly appreciated! The easiest way to get involved is through the GitHub discussions page or on [discord](https://discord.gg/dCMfwuze48)!

### [spindalis](https://github.com/lignum-vitae/spindalis)

Spindalis is a bioinformatics-focused library for numerical modeling, optimization, data analysis, and simulation written. This library is still in early-stage development but is intended to make working with algorithms commonly used in bioinformatics (and other areas) as simple to use as possible.

There are issues posted for the next items to be added to the library. If you're interested in helping, don't hesitate to check out the repository! Any help, ideas, or questions are welcome!

